### ISSUE TITLE
Invalid attribute location warning suppresses errors

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#60739]
Compile this code:

```cs
[a: a]
class C {}
```

#### Expected

Some kind of error.

#### Actual

Code compiles with a warning.

**Comment by 333fred** [comment]
Note that there are two issues here:
1. Total nonsense attribute locations are accepted.
2. We don't do any binding of the thing specified in the attribute location to validate it is a thing that is real.

**Comment by davidwengier** [comment]
Thats a funny way of saying "We have an undocumented comment syntax we should document"

**Comment by svick** [comment]
> Total nonsense attribute locations are accepted.

This is correct according to [the spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/attributes#213-attribute-specification):

> An implementation can accept other *attribute_target*s, the purposes of which are implementation defined. An implementation that does not recognize such an *attribute_target* shall issue a warning and ignore the containing *attribute_section*.

It's not clear to me if "ignore" here includes not checking whether the attribute type even exists, but it seems the compiler interprets the spec this way.

**Comment by Youssef1313** [comment]
@333fred Is this by-design per @svick comment?

**Comment by 333fred** [comment]
Depends on whether we decide that the compiler is correctly interpreting `ignore`. IMO, it's not.
